monitoring/uss_qualifier/scenarios/astm/utm/subscription_notifications/test_steps/validate_notification_received.py

```diff
@@ -0,0 +1,164 @@
+    This step checks if expected notification was received by tested_uss from mock_uss.
+    As it could take time for the notification to be sent, this function will poll for the notifications,
+    till a max time is reached.
+    Args:
+        interactions_since_time: the earliest time a notification may have been sent
+        op_intent_ref_id: expected operational_intent_id in the notification
+        subscription_id: expected subscription_id in the notification
+        tested_uss_base_url: base_url of tested_uss to which the notification should be sent
+        tested_uss_participant_id: participant_id of the tested_uss receiving the notification
+        plan_request_time: timestamp of the mock_uss flight plan query that would lead to sending notification
+    """
+
+    # Check for Mock USS interactions (with notifications) in intervals till max wait time reached
+    interactions, query = wait_in_intervals(mock_uss_interactions)(
+        scenario=scenario,
+        mock_uss=mock_uss,
+        op_id=OperationID.NotifyOperationalIntentDetailsChanged,
+        direction=QueryDirection.Outgoing,
+        since=StringBasedDateTime(interactions_since_time),
+        is_applicable=_is_notification_sent_to_url_with_op_intent_id(
+            op_intent_ref_id, tested_uss_base_url
+        ),
+    )
+
+    #  Check if a notification exists with expected subscription_id, in the interactions
+    (
+        notification_with_subscr_id_sent,
+        resp_status,
+    ) = _check_notification_exists_with_subscription_id(interactions, subscription_id)
+
+    with scenario.check(
+        "Mock USS sends valid notification", mock_uss.participant_id
+    ) as check:
+        if not interactions:
+            check.record_failed(
+                summary=f"No notification sent to tested_uss",
+                details=f"Notification to tested_uss with pre-existing relevant operational intent not sent even though DSS instructed mock_uss to notify due to subscription.",
+                query_timestamps=[plan_request_time, query.request.timestamp],
+            )
+        if interactions and not notification_with_subscr_id_sent:
+            check.record_failed(
+                summary=f"Invalid notification sent by mock_uss",
+                details=f"Invalid notification sent by mock_uss to tested_uss - invalid format or missing subscription_id ({subscription_id}), of the operational intent owned by tested_uss .",
+                query_timestamps=[plan_request_time, query.request.timestamp],
+            )
+
+    if notification_with_subscr_id_sent and resp_status != 204:
+        with scenario.check(
+            "Tested USS receives valid notification", [tested_uss_participant_id]
+        ) as check:
+            check.record_failed(
+                summary=f"Valid notification not accepted by tested_uss.",
+                details=f"Valid notification by mock_uss not accepted by tested_uss. Tested_uss should have responded with status 200.",
```

Comment:
```suggestion
                details=f"Valid notification by mock_uss not accepted by tested_uss. Tested_uss should have responded with status 204.",
```
Also it'd be useful to put in details the status code actually returned.